synapseclient/api/configuration_services.py

```diff
@@ -76,24 +77,46 @@ def get_client_authenticated_s3_profile(
         section_name=config_section, config_path=config_path
     ).get("profile_name", "default")
 
-
-def get_config_authentication(
-    config_path: str,
-) -> Dict[str, str]:
+def get_config_authentication(config_path: str, profile: str = "default") -> Dict[str, str]:
     """
     Get the authentication section of the configuration file.
+    Reads the Synapse configuration file and extracts authentication information
+    for a given profile.
 
-    Arguments:
-        config_path:  Path to configuration file on local file system
+    Args:
+        config_path (str): Path to the Synapse config file.
+        profile (str, optional): The profile name to retrieve credentials for. Defaults to "default".
 
     Returns:
-        The authentication section of the configuration file
+        Dict[str, str]: Dictionary containing 'username' and 'auth_token'.
+
+    Raises:
+        SynapseNoCredentialsError: If the specified profile is not found in the config file.
+        FileNotFoundError: If the config file does not exist.
+
+    Example:
+        >>> get_config_authentication("~/.synapseConfig", profile="user1")
+        {'username': 'user1', 'auth_token': 'user1_auth_token'}
     """
-    return get_config_section_dict(
-        section_name=config_file_constants.AUTHENTICATION_SECTION_NAME,
-        config_path=config_path,
-    )
+    config = configparser.ConfigParser()
+    config.read(config_path)
+
+    auth_profiles = {}
+
+    # Extract all profiles and normalize them
+    for section in config.sections():
+        normalized_profile = section.replace("profile ", "").strip()  # Ensure no trailing spaces
+        auth_profiles[normalized_profile] = {
+            "username": config.get(section, "username", fallback=None),
+            "auth_token": config.get(section, "authtoken", fallback=None),
+        }
+
+    # **Check for the exact match**
+    if profile in auth_profiles:
+        return auth_profiles[profile]
 
+    # If profile not found, return error instead of "default"
+    raise SynapseNoCredentialsError(f"Profile '{profile}' not found in the config file.")
```

Comment:
The code you're doing here isn't needed and we shouldn't be modifying potential behavior. See my comment I left: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1183#discussion_r1999225453 on what I think the code should be instead.